core/aws-core/src/main/java/software/amazon/awssdk/awscore/AwsServiceClientConfiguration.java

```diff
@@ -113,6 +116,17 @@ public final Region region() {
         public final URI endpointOverride() {
             return endpointOverride;
         }
+
+        @Override
+        public final EndpointProvider endpointProvider() {
+            return endpointProvider;
+        }
+
+        @Override
+        public Builder endpointProvider(EndpointProvider endpointProvider) {
```

Comment:
should we put this Builder override in Builder instead of BuilderImpl to keep consistent with other fields?